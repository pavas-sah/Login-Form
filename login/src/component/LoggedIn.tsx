import React from 'react';
import { connect } from "react-redux";
import { RouteComponentProps } from "react-router-dom";

interface IProps extends RouteComponentProps{
    onValidUser(): any,
    onInvalidUser(): any
   }

const LoggedIn = (props:IProps) => {
    return (
        <div>
            <button onClick = {props.onInvalidUser}>LogOut</button>
            User Logged In Successfully.
        </div>
    )
    
}

const mapDispatchToProps = (dispatch: any) => {
    return {
        onInvalidUser: () => dispatch({ type: 'LOGOUT' }),
    };
};

export default connect(null, mapDispatchToProps)(LoggedIn); 