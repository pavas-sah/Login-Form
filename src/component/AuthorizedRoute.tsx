import { Redirect, Route } from "react-router-dom"
import React from "react"
import { IGlobalState } from "../store/LoginReducer";

export interface IProps {
    path: any,
    component: any,
}

export class AuthorizedRoute extends React.Component<IProps & IGlobalState>{
    
    render() {
        const { component: Component, ...rest } = this.props
        return (
            <Route  {...rest} render={() => {
                return this.props.isAuthenticated !== undefined && this.props.isAuthenticated
                    ? < Component data-testid="comp" {...this.props} />
                    : <Redirect to="/" />
            }} />
        )
    }
}

