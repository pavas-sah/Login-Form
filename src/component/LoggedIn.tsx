import React, { Dispatch } from 'react';
import { Action } from 'redux';

export interface IProps {
    onInvalidUser(): Dispatch<Action>
   }

export const LoggedIn = (isValidUser:(IProps)) => {
    return (
        <div>
            <button data-testid="button" onClick = {isValidUser.onInvalidUser}>LogOut</button>
            User Logged In Successfully.
        </div>
    )
    
}
