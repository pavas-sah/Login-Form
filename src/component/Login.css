.main-container{
    padding-top: 200px;
}
.error{
    color: red;
}