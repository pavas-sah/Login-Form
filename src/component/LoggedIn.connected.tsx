import { connect } from "react-redux";
import { LoggedIn } from "./LoggedIn";
import { LoginState } from "../store/LoginReducer.action";

export const mapDispatchToProps = (dispatch: any) => {
    return {
        onInvalidUser: () => dispatch({ type: LoginState.LOGOUT }),
    };
};

export default connect(null, mapDispatchToProps)(LoggedIn); 