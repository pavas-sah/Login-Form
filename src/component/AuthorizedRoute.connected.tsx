import { IGlobalState } from "../store/LoginReducer";
import { AuthorizedRoute } from "./AuthorizedRoute";
import { connect } from "react-redux";

const mapStateToProps = (globalState: IGlobalState) => {
    return {
        isAuthenticated : globalState.isAuthenticated,
    };
};

export default connect(mapStateToProps)(AuthorizedRoute)