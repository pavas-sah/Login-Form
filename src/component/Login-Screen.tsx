import React, { Component } from "react";
import { Redirect, RouteComponentProps } from "react-router-dom";
import "./Login.css";
import { connect } from "react-redux";

export interface IState {
    user: string,
    pass: string,
    toSuccess: boolean,
    isUserValid: boolean,
    isPassValid: boolean,
    errorMessage: string
}

export interface IProps extends RouteComponentProps {
    onValidUser(): any,
    onInvalidUser(): any
}

export class LoginScreen extends Component<IProps, IState> {
    state: IState = {
        user: '',
        pass: '',
        toSuccess: false,
        isPassValid: true,
        isUserValid: true,
        errorMessage: ""
    };

    // componentWillUpdate()
    // {
    //     debugger;
    //     return true;

    // }

    validEmailRegex = RegExp(
        // eslint-disable-next-line
        /^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i
    );
    validPasswordRegex =
        RegExp(/^[0-9a-zA-Z]*$/);

    handleUserId = (e: any) => {

        // let errors = this.state.errors;
        this.setState({
            user: e.target.value,
            errorMessage: "",
            isUserValid: this.validEmailRegex.test(e.target.value),
        });
    }

    handlePassword = (e: any) => {

        // let errors = this.state.errors;
        this.setState({
            pass: e.target.value,
            isPassValid: this.validPasswordRegex.test(e.target.value),
            errorMessage: ""
        });
    }

    formSubmitted = (form: React.FormEvent<HTMLFormElement>) => {
        form.preventDefault();
        if (this.state.user === "pavas.sah@gmail.com" && this.state.pass === "pass") {
            this.setState({
                toSuccess: true,
                errorMessage: ""
            })
            this.props.onValidUser();
        } else {
            this.setState({ errorMessage: "Invalid User Id/Password" });
        }
    }


    haveErrors = () => {
        let haveError: boolean = false;
        if (this.state.user === '' || this.state.pass === '') {
            haveError = true;
        }

        if (!this.state.isPassValid || !this.state.isUserValid) {
            haveError = true;
        }
        return haveError;
    }


    render() {

        if (this.state.toSuccess === true) {
            return <Redirect to='/logged-in' />
        }

        return (
            <div className="main-container">

                <form onSubmit={this.formSubmitted}>
                    <div>
                        <label>User Name :</label>
                        <input type="text" id="user" onChange={this.handleUserId} data-validation-type="email" required />
                        {!this.state.isUserValid &&
                            <label className="error">User Id entered is not a valid email id.</label>
                        }
                    </div>
                    <div>
                        <label>Password :</label>
                        <input type="password" id="pass" onChange={this.handlePassword} required />
                        {!this.state.isPassValid &&
                            <label className="error">Password should be alphanumeric.</label>
                        }
                    </div>
                    <button disabled={this.haveErrors()} >Login</button>
                    <div className="error">{this.state.errorMessage}</div>
                </form>

            </div>
        );
    }
}

const mapDispatchToProps = (dispatch: any) => {
    return {
        onValidUser: () => dispatch({ type: 'LOGIN' }),
        // onInvalidUser: () => dispatch({ type: 'LOGOUT' }),
    };
};

export default connect (null, mapDispatchToProps)(LoginScreen);

