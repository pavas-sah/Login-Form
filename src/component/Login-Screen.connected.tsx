import { connect } from "react-redux";
import { LoginScreen} from "./Login-Screen";
import { IGlobalState } from "../store/LoginReducer";
import { LoginState } from "../store/LoginReducer.action";

export const mapStateToProps = (globalState: IGlobalState) => {
    return {
        isAuthenticated : globalState.isAuthenticated,
        push : props.history.push
    };
};

export const mapDispatchToProps = (dispatch: any) => {
    return {
        onValidUser: () => dispatch({ type: LoginState.LOGIN }),
    };
};

export default connect (mapStateToProps, mapDispatchToProps)(LoginScreen);
