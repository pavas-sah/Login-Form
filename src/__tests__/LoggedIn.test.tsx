import React, { Component } from 'react';
import renderer from 'react-test-renderer';
import { LoggedIn, IProps } from '../component/LoggedIn';
import { render } from '@testing-library/react';


const mockInvalidUser = jest.fn();

const loggedInProps: IProps = {
    onInvalidUser : mockInvalidUser
  };

describe('LoggedIn Page',()=>{

    it('renders correctly', ()=>{
        const { queryByTestId } = render(<LoggedIn {...loggedInProps} />);
        expect(queryByTestId("button")).toBeTruthy();
    })

    it('matches snapshot', () =>{
        const tree = renderer.create(<LoggedIn {...loggedInProps}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

});