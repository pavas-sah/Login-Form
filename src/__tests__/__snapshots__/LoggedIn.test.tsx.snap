// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggedIn Page matches snapshot 1`] = `
<div>
  <button
    data-testid="button"
    onClick={[MockFunction]}
  >
    LogOut
  </button>
  User Logged In Successfully.
</div>
`;
